pkg/grafana/folders.go

```diff
@@ -159,6 +159,39 @@ func putFolder(resource grizzly.Resource) error {
 	return nil
 }
 
+var getFolderById = func(folderId int64) (Folder, error) {
+	grafanaURL, err := getGrafanaURL(fmt.Sprintf("folders/id/%d", folderId))
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := http.Get(grafanaURL)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusNotFound:
+		return nil, fmt.Errorf("NOT FOUND")
```

Comment:
I think we have an `ErrNotFound` we can use. I recognise that this isn't your doing thought - I'm sure this is wrong in other places too.